### ISSUE TITLE
USB & Monterey

### ISSUE THREAD

**Issue by Xodiced** [seven-of-eleven/Lenovo-ThinkPad-L13-Yoga-Hackintosh#11]
You can use this to boot Monterey and it will work except for the USB ports and booting with -v sees BT adapter errors adding 3(!) minutes to the boot time.

**Comment by seven-of-eleven** [comment]
I haven't installed Monterey yet. I included the Airportitlwm kext for it in anticipation of it coming out this month (hopefully) and then being able to install it.  I'm waiting for the official release. Once it is out I'll update my EFI and post instructions as necessary.

I'm surprised USB ports are causing issues as the ports are mapped and I haven't heard anything about changes to the port mapping process.  Let me know if you can determine the issue.  I have read that **IntelBluetoothInjector** isn't needed but again I haven't tried it yet.

Thanks for letting me know.

**Comment by seven-of-eleven** [comment]
Found this on another repo. Seems the bluetooth issue is known for Monterey. Hopefully there is a fix soon.

> You can enable IntelBluetoothFirmware.kext & IntelBluetoothInjector.kext to be able to turn off bluetooth by enabling those kexts in config.plist This is not done by default because it increases boot times for those on macOS Monterey

**Comment by seven-of-eleven** [comment]
Found info below as well as [instruction from site](https://openintelwireless.github.io/IntelBluetoothFirmware/FAQ.html#what-additional-steps-should-i-do-to-make-bluetooth-work-on-macos-monterey)

**For Monterey you need:**

• IntelBluetoothFirmware ONLY, do NOT use IntelBluetoothInjector

• BlueToolFixup

**For Bigsur and below:**

• IntelBluetoothFirmware

• IntelBluetoothInjector

**Comment by Xodiced** [comment]
thanks!  I'll give it a go and try to figure out the USB issue tonight.

**Comment by Xodiced** [comment]
Ok dropping intelbluetoothinjector and using bluetoolfixup has functioning usb and wifi (With the proper kext) on Monterey, though no working BT support.
![Screen Shot 2021-09-24 at 12 29 51 AM](https://user-images.githubusercontent.com/76649021/134618567-3f07b7ac-ad43-4228-a5ff-d514f63180e9.png)

**Comment by seven-of-eleven** [comment]
I'm assuming it fixed the boot times? 

Did you happen to check which version of IntelBluetooth* you were using. Only reason I ask is they mention 1.1.3. but I included version 2.0.0 in my EFI. 

I'm hopeful this will be fixed b/f the release of Monterey.  I'll keep you posted if I find anything new regarding this.  Thanks for your input and testing.

**Comment by Xodiced** [comment]
Sorry for the very late reply.

I did try 1.1.3 also as I read the same as you, but it made no change vs. 2.0.0.
And yes the steps I took did fix the boot times.

Now that Monterey is released I will be playing more and if I find anything I will post more info.

**Comment by Xodiced** [comment]
I just finished updating the EFI with the latest kexts and removing the unneeded bluetooth kext while adding the new one and everything works fine on Monterey now. But do not go past opencore 0.7.4 or you will get a black screen at boot.

TL;DR Monterey works fine with this EFI if you use the latest kexts and do the Bluetooth modifications stated above.
